AppCenter/AppCenter/Internals/Channel/MSLogManagerDefault.m

```diff
@@ -155,6 +155,16 @@ - (void)processLog:(id<MSLog>)log forGroupId:(NSString *)groupId {
     log.device = [[MSDeviceTracker sharedInstance] device];
   }
 
+  // Check if the logs should be filtered out.
+  __block BOOL shouldFilter = NO;
+  [self enumerateDelegatesForSelector:@selector(shouldFilterLog:)
+                            withBlock:^(id<MSLogManagerDelegate> delegate) {
+                              shouldFilter = shouldFilter || [delegate shouldFilterLog:log];
+                            }];
+  if (shouldFilter) {
+    return;
+  }
+
   // Notify delegates.
   [self enumerateDelegatesForSelector:@selector(onPreparedLog:withInternalId:)
```

Comment:
is that step doing the session id decoration? If yes need to move shouldFilter after that step and adapt unit tests to verify session id not missing if we filter out.